src/client/NextApi.Client/NextApiClient.cs

```diff
@@ -303,30 +316,40 @@ private async Task<T> InvokeHttp<T>(NextApiCommand command)
                 }
             }
 
-            switch (httpSerializationType)
+            var data = await response.Content.ReadAsStringAsync();
+            var result =
+                JsonConvert.DeserializeObject<NextApiResponseJsonWrapper<T>>(data, SerializationUtils.GetJsonConfig());
+
+            if (!result.Success)
             {
-                case SerializationType.Json:
-                {
-                    // process as normal nextapi response
-                    var data = await response.Content.ReadAsStringAsync();
-                    var result =
-                        JsonConvert.DeserializeObject<NextApiResponseJsonWrapper<T>>(data, SerializationUtils.GetJsonConfig());
-                    if (!result.Success)
-                        throw NextApiClientUtils.NextApiException(result.Error);
-
-                    return result.Data;
-                }
-                case SerializationType.MessagePack:
+                throw NextApiClientUtils.NextApiException(result.Error);
+            }
+
+            return result.Data;
+        }
+
+        private async Task<T> InvokeHttpMessagePack<T>(HttpResponseMessage response)
+        {
+            var mediaType = response.Content.Headers.ContentType.MediaType;
```

Comment:
There is already block with same definitions on lines 305-317. Both `InvokeHttpJson` and the `InvokeHttpMessagePack`  are being used in `InvokeHttp`. Perhaps, put this block in a separate method, for example in `NextApiUtils`, or put it in a calling method `InvokeHttp`, to not repeat same code over again.